### PR TITLE
feat: support custom os image import from COS into PowerVS workspace

### PR DESCRIPTION
**Pull request by ludwig-mueller**
### Description

Add feature to support custom os image import from COS into PowerVS workspace

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [X] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: variables.tf
Comment (surajsbharadwaj) [inline variables.tf:1]: variable name to be changed from custom_pi to pi_custom.
Set defaults here to [], {} or null depending on the type or use case you want to handle

**Review by surajsbharadwaj — APPROVED** [review]

**Review by surajsbharadwaj — CHANGES_REQUESTED** [review]
Please have a look at the comments. Minor changes

File: variables.tf [hunk 1]
```diff
@@ -107,3 +107,100 @@ variable "pi_tags" {
...
+    bucket_name   = string
+    bucket_access = string
+    bucket_region = string
+  })
+  default = null
+  validation {
+    condition     = var.pi_custom_image_cos_configuration != null ? contains(["public", "private"], var.pi_custom_image_cos_configuration.bucket_access) : true
+    error_message = "Invalid pi_custom_image_cos_configuration.bucket_access. Allowed values: [\"public\", \"private\"]."
+  }
+  validation {
```
Comment (surajsbharadwaj) [inline variables.tf:191]: I don't think this possible. Please check. Cannot access another variable inside validation. Will have to move this validation to local variable inside pi_images.tf file

**Review by surajsbharadwaj — CHANGES_REQUESTED** [review]
Remove the example folder. Not required. Use it locally for testing . Do not commit it to repo.

Also some minute changes

File: variables.tf (hunk 1 above)
Comment (ludwig-mueller) [reply variables.tf:203]: I made a logic error in the validation and fixed it. I tried it and it now validates correctly. If all images are null it doesn't throw an error but if at least one is defined, it throws an error.
Bildschirmfoto 2024-10-21 um 16 02 50

**Review by surajsbharadwaj — APPROVED** [review]

**Review by surajsbharadwaj — APPROVED** [review]

**Pull request merged**